frontend/src/v5/store/tickets/tickets.helpers.ts

```diff
@@ -15,8 +15,135 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+import { formatMessage } from '@/v5/services/intl';
 import { FederationsHooksSelectors } from '@/v5/services/selectorsHooks/federationsSelectors.hooks';
+import { trimmedString } from '@/v5/validation/shared/validators';
+import { isEmpty, isUndefined } from 'lodash';
+import * as Yup from 'yup';
+import { PropertyDefinition } from './tickets.types';
 
 export const modelIsFederation = (modelId: string) => (
 	!!FederationsHooksSelectors.selectContainersByFederationId(modelId).length
 );
+
+const MAX_TEXT_LENGTH = 50;
+const MAX_LONG_TEXT_LENGTH = 120;
+const maxStringLength = (type) => (type === 'longText' ? MAX_LONG_TEXT_LENGTH : MAX_TEXT_LENGTH);
+
+export const propertyValidator = ({ required, name, type }: PropertyDefinition) => {
+	let validator;
+	const maxLength = maxStringLength(type);
+
+	switch (type) {
+		case 'text':
+		case 'longText':
+			validator = trimmedString.max(maxLength,
+				formatMessage({
+					id: 'validation.ticket.tooLong',
+					defaultMessage: ' {name} is limited to {maxLength} characters',
+				},
+				{ maxLength, name }));
+			break;
+		case 'coords':
+			validator = Yup.array();
+			break;
+		case 'manyOf':
+			validator = Yup.array();
+			break;
+		case 'oneOf':
+			validator = trimmedString;
+			break;
+		case 'boolean':
+			validator = Yup.boolean();
+			break;
+		case 'number':
+			validator = Yup.number().nullable(true).transform((_, val) => Number(val) || null);
+			break;
+		case 'date':
+			validator = Yup.date().nullable();
+			break;
+		default: validator = trimmedString;
+	}
+
+	if (required) {
+		if (type === 'manyOf') {
+			validator = validator.min(1,
+				formatMessage({
+					id: 'validation.ticket.manyOf.required',
+					defaultMessage: 'Select at least one option',
+				}));
+		}
+		if (type === 'coords') {
+			const requiredCoord = Yup.string().required(
+				formatMessage({
+					id: 'validation.ticket.coord.required',
+					defaultMessage: 'This is required',
+				}),
+			);
+			validator = Yup.array(requiredCoord);
+		}
+		validator = validator.required(
+			formatMessage({
+				id: 'validation.ticket.requiredField',
+				defaultMessage: '{name} is a required field',
+			},
+			{ name }),
+		);
+	}
+
+	return validator;
+};
+
+export const propertiesValidator = (properties) => {
+	const validators = properties.reduce(
+		(validatorsObj, property) => ({
+			...validatorsObj,
+			[property.name]: propertyValidator(property),
+		}),
+		{},
+	);
+	return Yup.object().shape(validators);
+};
+
+const parseModule = (module) => {
+	const parsedModule = {};
+	Object.entries(module)
+		// skip nullish values that are not 0s or false
+		.filter((entry) => ![null, undefined, ''].includes(entry[1] as any))
+		.forEach(([key, value]) => {
+			if (Array.isArray(value)) {
+				if (value.length === 0) return;
+				// manyOf should not allow empty or nullish values, but coords may.
```

Comment:
I could be wrong here, but I think manyOf should allow empty values? With the assignees the value can be an empty array to show that it is part of the template but no assignees have been assigned yet. In this case the assignees list component will display 'Unassigned' , whereas if the assignees doesn't exist on the properties it will not be rendered.

A similar thing also happens for the dueDate, where it can be null and the ticket list item will show that the ticket can have a due date but it's jut unset.

@sanmont3drepo do you have any thoughts on this? Alternatively the ticket list item could check if the ticket's template includes assignees/dueDate but they're missing in the ticket, and then render the empty states

Reply:
>  but I think manyOf should allow empty values?

Yes it should allow empty values unless required is set to true

Reply:
in fact `coords` does not allow empty or nullish values so that comment is very wrong 😆 

For coords you either don't have it (again, if required is set to true), or it has to be [x, y, z]

Reply:
See [here](https://github.com/3drepo/3drepo.io/blob/staging/backend/src/v5/schemas/tickets/validators.js#L89), Coords is of `type.position` which is of[ type array of numbers with length 3](https://github.com/3drepo/3drepo.io/blob/staging/backend/src/v5/utils/helper/yup.js#L70)

Reply:
I now understand that my comment is not as clear as I thought. I'll rephrase it here and then fix the comment in the code.

What I mean by that is that manyOf takes a value, which may indeed be `undefined` (and if so, it is filter right before the forEach)
```ts
Object.entries(module)
	// skip nullish values that are not 0s or false
	.filter((entry) => ![null, undefined, ''].includes(entry[1] as any)) // <- This line here
	.forEach(([key, value]) => {
```
or an empty array (which would also not be a problem)
```ts
if (Array.isArray(value)) {
	if (value.length === 0) return;  // <- This line here
```
However, when its value is a non empty array, the elements of that array should each represent the value of a selected option from the dropdown menu. Those very values should not be undefined